community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java

```diff
@@ -289,6 +295,14 @@ boolean applicable( DiagnosticsPhase phase )
     }
 
     public static final String DEFAULT_DATA_SOURCE_NAME = "nioneodb";
+
+    /**
+     * This setting is hidden to the user and is here merely for making it easier to back out of
+     * a change where reading property chains incurs read locks on {@link LockService}.
+     */
+    private static final Setting<Boolean> use_read_locks_on_property_reads =
+            setting( "experimental.use_read_locks_on_property_reads", BOOLEAN, TRUE );
+
```

Comment:
It's looked up and parsed once per database so shouldn't make a difference in that sense. Having it as a system property would also work because it's not user-facing anyhow. Doesn't matter to me which we choose.